lib/smack/MemorySafetyChecker.cpp

```diff
@@ -3,30 +3,63 @@
 //
 #include "smack/MemorySafetyChecker.h"
 #include "smack/Naming.h"
+#include "smack/SmackOptions.h"
 #include "llvm/IR/Module.h"
 #include "llvm/IR/Instructions.h"
 #include "llvm/IR/InstIterator.h"
 #include "llvm/IR/Constants.h"
 #include "llvm/Support/Debug.h"
 #include "llvm/IR/ValueSymbolTable.h"
+#include "llvm/IR/IntrinsicInst.h"
 
 namespace smack {
 
 using namespace llvm;
 
+void insertMemoryLeakStatement(Function& F, Module& m) {
+  Function* memoryLeakCheckFunction = m.getFunction(Naming::MEMORY_LEAK_CHECK_FUNCTION);
```

Comment:
Add:
`assert (memoryLeakCheckFunction != NULL && "Memory leak check function must be present.");`